#VRML_SIM R2022a utf8
WorldInfo {
}
Viewpoint {
  orientation -0.3345400833333094 0.3094103042510244 0.8901394251838478 1.7600100792918496
  position -0.4316575687745271 -6.070154130699884 4.145117031919176
}
TexturedBackground {
}
TexturedBackgroundLight {
}
RectangleArena {
  translation 0 -0.59 -0.04
  floorSize 10 10
  floorAppearance PBRAppearance {
    metalness 0
  }
}
Wall {
  translation -0.03191 -0.8205 -0.04
  name "wall 1"
  size 1.5 0.05 0.1
  appearance Roughcast {
    colorOverride 1 0 0
    textureTransform TextureTransform {
      scale 2.4 1
    }
  }
}
Wall {
  translation -1.5 -0.627305 -0.04
  rotation 0 0 1 2.8799003061004256
  name "wall 2"
  size 1.5 0.05 0.1
  appearance Roughcast {
    colorOverride 1 0 0
    textureTransform TextureTransform {
      scale 2.4 1
    }
  }
}
Wall {
  translation -2.21404 -0.819153 -0.04
  rotation 0 0 1 2.8799003061004256
  name "wall 3"
  size 0.2 0.05 0.1
  appearance Roughcast {
    colorOverride 1 0 0
    textureTransform TextureTransform {
      scale 2.4 1
    }
  }
}
CurvedRoadSegment {
  translation -0.69 -0.83 -0.0399
  name "road 1"
  width 0.11
  numberOfLanes 1
  rightBorder FALSE
  leftBorder FALSE
  curvatureRadius 2
  appearance DEF black_road PBRAppearance {
    baseColor 0 0 0
    metalness 0
  }
  pavementAppearance Pavement {
    type "black stone"
  }
  bottomTexture [
    ""
  ]
  locked FALSE
  roadBoundingObject TRUE
}
CurvedRoadSegment {
  translation -0.69 -0.83 -0.0399
  rotation 0 0 -1 -1.5707953071795862
  name "road 2"
  width 0.11
  numberOfLanes 1
  rightBorder FALSE
  leftBorder FALSE
  curvatureRadius 2
  appearance USE black_road
  locked FALSE
  roadBoundingObject TRUE
}
CurvedRoadSegment {
  translation -0.69 -0.83 -0.0399
  rotation 0 0 1 3.14159
  name "road 3"
  width 0.11
  numberOfLanes 1
  rightBorder FALSE
  leftBorder FALSE
  curvatureRadius 2
  appearance USE black_road
  locked FALSE
}
Robot {
  translation -0.57208 -2.82355 0.0166916
  rotation -0.016235594159476598 -0.014528194773689173 0.9997626403496516 0.0936886
  children [
    DistanceSensor {
      translation -1.48212 2.19078 0
      rotation 0 0 1 1.2
      name "ds_detect"
      type "infra-red"
    }
    DistanceSensor {
      translation -0.15 0.03 -0.02
      rotation 0 1 0 1.57
      children [
        DEF ds_shape Shape {
          appearance PBRAppearance {
            baseColor 1 0.33 0
            metalness 0
          }
          geometry Box {
            size 0.01 0.01 0.01
          }
        }
      ]
      name "ds_right"
      boundingObject USE ds_shape
      physics DEF ds_physics Physics {
      }
      locked TRUE
      type "infra-red"
    }
    DistanceSensor {
      translation -0.15 -0.03 -0.02
      rotation 0 1 0 1.57
      children [
        DEF ds_shape Shape {
          appearance PBRAppearance {
            baseColor 1 0.33 0
            metalness 0
          }
          geometry Box {
            size 0.01 0.01 0.01
          }
        }
      ]
      name "ds_left"
      boundingObject USE ds_shape
      physics DEF ds_physics Physics {
      }
      locked TRUE
      type "infra-red"
    }
    HingeJoint {
      jointParameters HingeJointParameters {
        position 16.216686401762963
        axis 0 -1 0
        anchor -0.07 -0.09 0.01
      }
      device [
        RotationalMotor {
          name "front_left_motor"
        }
      ]
      endPoint Solid {
        translation -0.07 -0.09 0.01
        rotation 0.18074302747404025 -0.6957378350700197 -0.6951840208690073 3.499496619847691
        children [
          DEF wheel_shape Shape {
            appearance PBRAppearance {
              baseColor 1 0.33 1
            }
            geometry Cylinder {
              height 0.03
              radius 0.06
            }
          }
        ]
        name "solid(3)"
        boundingObject USE wheel_shape
        physics DEF wheel_physics Physics {
        }
        locked TRUE
      }
    }
    HingeJoint {
      jointParameters HingeJointParameters {
        position 33.53455288142623
        axis 0 -1 0
        anchor -0.07 0.09 0.01
      }
      device [
        RotationalMotor {
          name "front_right_motor"
        }
      ]
      endPoint Solid {
        translation -0.07 0.09 0.01
        rotation -0.34626183573315805 0.6633204383511241 -0.663406916741434 3.8081935082583644
        children [
          DEF wheel_shape Shape {
            appearance PBRAppearance {
              baseColor 1 0.33 1
            }
            geometry Cylinder {
              height 0.03
              radius 0.06
            }
          }
        ]
        name "solid(2)"
        boundingObject USE wheel_shape
        physics DEF wheel_physics Physics {
        }
        locked TRUE
      }
    }
    HingeJoint {
      jointParameters HingeJointParameters {
        position 35.741533897717304
        axis 0 -1 0
        anchor 0.07 0.09 0.01
      }
      device [
        RotationalMotor {
          name "rear_right_motor"
        }
      ]
      endPoint Solid {
        translation 0.07 0.09 0.01
        rotation -0.34611608511614467 0.6636657989175472 -0.663137514374664 3.808506238828619
        children [
          DEF wheel_shape Shape {
            appearance PBRAppearance {
              baseColor 1 0.33 1
            }
            geometry Cylinder {
              height 0.03
              radius 0.06
            }
          }
        ]
        name "solid(1)"
        boundingObject USE wheel_shape
        physics DEF wheel_physics Physics {
        }
        locked TRUE
      }
    }
    HingeJoint {
      jointParameters HingeJointParameters {
        position 20.267249684006686
        axis 0 -1 0
        anchor 0.07 -0.09 0.01
      }
      device [
        RotationalMotor {
          name "rear_left_motor"
        }
      ]
      endPoint Solid {
        translation 0.07 -0.09 0.01
        rotation 0.21945954695872374 -0.6901432752986775 -0.689593914422601 3.573995724545174
        children [
          DEF wheel_shape Shape {
            appearance PBRAppearance {
              baseColor 1 0.33 1
            }
            geometry Cylinder {
              height 0.03
              radius 0.06
            }
          }
        ]
        boundingObject USE wheel_shape
        physics DEF wheel_physics Physics {
        }
        locked TRUE
      }
    }
    DEF robo_shape Shape {
      appearance PBRAppearance {
        baseColor 0 1 1
        metalness 0
      }
      geometry Box {
        size 0.3 0.14 0.04
      }
    }
  ]
  name "line_follower"
  boundingObject USE robo_shape
  physics DEF robo_physics Physics {
  }
  controller "LineFollower"
}
Robot {
  translation 0.743053 -0.98615 0.0166916
  children [
    DistanceSensor {
      translation -0.15 0.07 -0.02
      rotation 0.7071067811865475 0.7071067811865475 0 3.14
      children [
        DEF ds_shape Shape {
          appearance PBRAppearance {
            baseColor 1 0.33 0
            metalness 0
          }
          geometry Box {
            size 0.01 0.01 0.01
          }
        }
      ]
      name "ds_right"
      boundingObject USE ds_shape
      physics DEF ds_physics Physics {
      }
      locked TRUE
      type "sonar"
    }
    DistanceSensor {
      translation 0 -0.07 -0.02
      rotation 0.7071067811865475 -0.7071067811865475 0 3.14
      children [
        DEF ds_shape Shape {
          appearance PBRAppearance {
            baseColor 1 0.33 0
            metalness 0
          }
          geometry Box {
            size 0.01 0.01 0.01
          }
        }
      ]
      name "ds_left"
      boundingObject USE ds_shape
      physics DEF ds_physics Physics {
      }
      locked TRUE
      type "sonar"
    }
    HingeJoint {
      jointParameters HingeJointParameters {
        position 16.216686401762963
        axis 0 -1 0
        anchor -0.07 -0.09 0.01
      }
      device [
        RotationalMotor {
          name "front_left_motor"
        }
      ]
      endPoint Solid {
        translation -0.07 -0.09 0.01
        rotation 0.18074302747404025 -0.6957378350700197 -0.6951840208690073 3.499496619847691
        children [
          DEF wheel_shape Shape {
            appearance PBRAppearance {
              baseColor 1 0.33 1
            }
            geometry Cylinder {
              height 0.03
              radius 0.06
            }
          }
        ]
        name "solid(3)"
        boundingObject USE wheel_shape
        physics DEF wheel_physics Physics {
        }
        locked TRUE
      }
    }
    HingeJoint {
      jointParameters HingeJointParameters {
        position 33.53455288142623
        axis 0 -1 0
        anchor -0.07 0.09 0.01
      }
      device [
        RotationalMotor {
          name "front_right_motor"
        }
      ]
      endPoint Solid {
        translation -0.07 0.09 0.01
        rotation -0.34626183573315805 0.6633204383511241 -0.663406916741434 3.8081935082583644
        children [
          DEF wheel_shape Shape {
            appearance PBRAppearance {
              baseColor 1 0.33 1
            }
            geometry Cylinder {
              height 0.03
              radius 0.06
            }
          }
        ]
        name "solid(2)"
        boundingObject USE wheel_shape
        physics DEF wheel_physics Physics {
        }
        locked TRUE
      }
    }
    HingeJoint {
      jointParameters HingeJointParameters {
        position 35.741533897717304
        axis 0 -1 0
        anchor 0.07 0.09 0.01
      }
      device [
        RotationalMotor {
          name "rear_right_motor"
        }
      ]
      endPoint Solid {
        translation 0.07 0.09 0.01
        rotation -0.34611608511614467 0.6636657989175472 -0.663137514374664 3.808506238828619
        children [
          DEF wheel_shape Shape {
            appearance PBRAppearance {
              baseColor 1 0.33 1
            }
            geometry Cylinder {
              height 0.03
              radius 0.06
            }
          }
        ]
        name "solid(1)"
        boundingObject USE wheel_shape
        physics DEF wheel_physics Physics {
        }
        locked TRUE
      }
    }
    HingeJoint {
      jointParameters HingeJointParameters {
        position 20.267249684006686
        axis 0 -1 0
        anchor 0.07 -0.09 0.01
      }
      device [
        RotationalMotor {
          name "rear_left_motor"
        }
      ]
      endPoint Solid {
        translation 0.07 -0.09 0.01
        rotation 0.21945954695872374 -0.6901432752986775 -0.689593914422601 3.573995724545174
        children [
          DEF wheel_shape Shape {
            appearance PBRAppearance {
              baseColor 1 0.33 1
            }
            geometry Cylinder {
              height 0.03
              radius 0.06
            }
          }
        ]
        boundingObject USE wheel_shape
        physics DEF wheel_physics Physics {
        }
        locked TRUE
      }
    }
    DEF robo_shape Shape {
      appearance PBRAppearance {
        baseColor 1 1 0
        metalness 0
      }
      geometry Box {
        size 0.3 0.14 0.04
      }
    }
  ]
  name "wall_follower"
  boundingObject USE robo_shape
  physics DEF robo_physics Physics {
  }
  controller "WallFollower"
}
